src/AzureStorageQueueWorkerOptions.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Likvido.Worker.AzureStorageQueue
+{
+    public class AzureStorageQueueWorkerOptions
+    {
+        private TimeSpan _noMessagesSleepDuration = TimeSpan.FromSeconds(30);
+        private string _queueName;
+        private string _azureStorageConnectionString;
+        private TimeSpan _visibilityTimeout = TimeSpan.FromSeconds(30);
+        private string _poisonQueueName;
+        private int _maxRetryCount = 5;
+
+        /// <summary>
+        /// Default 30 seconds which is default for QueueClient.ReceiveMessagesAsync
+        /// </summary>
+        public TimeSpan VisibilityTimeout
+        {
+            get { return _visibilityTimeout; }
+            set
+            {
+                if (TimeSpan.FromSeconds(5) > value)
+                {
+                    throw new ArgumentException("Timeout must be 5 seconds minimum", nameof(VisibilityTimeout));
+                }
+                _visibilityTimeout = value;
+            }
+        }
+
+        [NotNull]
```

Comment:
If you do not set `QueueName` when you create a new instance of `AzureStorageQueueWorkerOptions`, then it would be null, right? - so why are you marking it with this attribute? (most likely I am missing something here)

This also reminds me that we should enable the C# 8 nullable checks for this project: https://stackoverflow.com/a/56267555/310937

Reply:
1. this attribute shows that the value must be provided. When you are consuming the library you can hit F12 and you'll this attribute.
2. I'll give nullable checks a try. Makes a lot of sense

Reply:
ok, so if you do:

```
var options = new AzureStorageQueueWorkerOptions();
var queueName = options.QueueName;
```

will you get a compiler warning or? (because in this case you have not set `QueueName`)

From what I can read about the NotNullAttribute, it just tells a consumer of the property that the value will never be null (even though the type allows null) ... but in my example above, the `queueName` would be null

Reply:
I'm thinking about this 
https://stackoverflow.com/a/32806406

Reply:
Ok, I don't really see the value of that attribute. I think we should remove it, and instead enable C# 8 nullable checks